src/odra_toml.rs

```diff
@@ -4,17 +4,36 @@ use std::path::{Path, PathBuf};
 
 use serde_derive::{Deserialize, Serialize};
 
-use crate::{command, errors::Error};
+use crate::{
+    command,
+    errors::{Error, Error::MalformedFqn},
+};
 
 /// Struct describing contract.
 #[derive(Deserialize, Serialize, Debug, Clone)]
 pub struct Contract {
-    /// Name of the contract
-    pub name: String,
-    /// Fully Qualified Name of the contract struct
     pub fqn: String,
 }
 
+impl Contract {
+    /// Extracts first part from fqn
+    pub fn module_name(&self) -> String {
+        self.fqn
+            .split_terminator("::")
+            .next()
```

Comment:
Does it work corectly, if the fqn has 3+ segments? For instance aa::bb::cc::Contract.